NewDawn/NewDawn/MainPageViewController.swift

```diff
@@ -16,17 +16,19 @@ class MainPageViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        if ProfileIndexUtil.loadProfileIndex() >= USER_DUMMY_DATA.count {
-            // TODO: When the user has seen all profiles, we go back to the first profile.
-            // In the future, we should go to an ending page
-            self.localStoreKeyValue(key: MAIN_PAGE_PROFILE_INDEX, value: 0)
+        self.getUserData(){ () -> () in
```

Comment:
It seems that each time user click skip, it will re-fetch profiles from backend. Ideally we should only fetch once when the main page is on daily fresh.
You can do that by moving this function to checkMainPageReload() in ViewerController.swift, and store the entire list of UserProfiles into local storage. You will then only need to read profiles from local storage.

Reply:
Will ViewController be too heavy if I move all fetching user data logic to it?

Reply:
Yes. But we can refactor that later

Reply:
Ideally, the processSessionTask should be in HttpUtil also. If you want, You can copy that function to HttpUtil for now. 

Then You can try put getUserData, getProfileRequest etc. into UserProfile class as a static method. Create another static method called buildUserProfile, which takes a request as input, and then return a list of UserProfile object as output. Then you use that function like UserProfile.buildUserProfile(request..) in changeMainPageReload() to fetch user info and make them as UserProfile objects and store them into local storage.

Reply:
Where do you want to put the UserProfileBuilder class?

Reply:
UserModel.swift would be fine